docs/scope-and-objectives.md

```diff
@@ -0,0 +1,167 @@
+
+#### Foundation
+
+- Cluster API MUST be able to be deployed and run on Kubernetes.
+
+- Cluster API MUST provide isolation between providers.
+
+- Cluster API MUST support multiple infrastructure providers, including both on-prem and cloud providers.
+
+- Cluster API MUST be able to manage groups/sets of Kubernetes nodes that support scaling and orchestrated upgrades.
+
+- Cluster API, through a single installation, MUST be able to manage (create, scale, upgrade, destroy) clusters on multiple providers.
+
+- Cluster API SHOULD support all operating systems in scope for Kubernetes conformance.
+
+#### User Experience
+
+- Cluster API MUST be able to display the versions of Kubernetes and related components of a Node that it manages.
+
+- Cluster API MUST be able to provide sufficient information for a consumer to access a provisioned Kubernetes cluster’s API server.
+
+#### Organization
+
+- Cluster API MUST NOT have code specific to a particular provider within the main repository, defined as sigs.k8s.io/cluster-api.
+
+#### Validation
+
+- Cluster API provider implementations SHOULD validate configuration data.
+
+- Cluster API MUST test integrations that it claims will work.
+
+- Cluster API MUST deploy clusters that pass Kubernetes conformance tests.
+
+- Cluster API MUST document how to bootstrap itself and MAY provide tooling to do so.
+
+- Cluster API MUST document requirements for machine images and SHOULD reference default tools to prepare machine images.
+
+- Cluster API MUST have conformance tests to ensure that Cluster API and a given provider fulfill the expectations of users/automation.
+
+#### Extension
+
+- Cluster API SHOULD allow default implementations to be pluggable/extensible.
+   - Cluster API SHOULD offer default implementations for generic operations.
+   - Users MAY use different provider implementations for different sets of operations.
+
+- 🔌 Cluster API MUST be able to provide information as to which versions of Kubernetes it can install.
+
+- 🔌 Cluster API MUST be able to bootstrap a new Kubernetes control plane.
+
+- 🔌 Cluster API MUST provide a generic image lookup mechanism that can filter on characteristics such as kubernetes version, container runtime, kernel version, etc.
+
+- 🔌 Cluster API MUST be able to provide a consistent (across providers) way to prepare, configure, and join nodes to a cluster.
+
+- 🔌 Cluster API MUST be able to scale a self-provisioned control plane up from and down to one replica.
+
+- 🔌 Cluster API MAY be able to scale an external control plane up from and down to one replica.
+
+- 🔌 Cluster API MUST provide high-level orchestration for Kubernetes upgrades (control plane and non-control plane).
+
+- 🔌 Cluster API MUST define and document configuration elements that it exclusively manages and take corrective actions if these elements are modified by external actors.
+
+- 🔌 Cluster API MUST provide the ability to receive the health status of a Kubernetes Node from an external source.
```

Comment:
External to what?

Reply:
external to cluster-api. This was added specifically to help facilitate external tooling providing higher level health checking for the cluster, such as the node-problem-detector.